src/python/WMCore/Storage/StageOutImpl.py

```diff
@@ -212,22 +217,28 @@ def __call__(self, protocol, inputPFN, targetPFN, options=None, checksums=None):
         # // Run the command
         # //
 
+        stageOutEx = None  # variable to store the possible StageOutError
         for retryCount in range(self.numRetries + 1):
             try:
                 logging.info("Running the stage out...")
                 self.executeCommand(command)
-                break
+                break  # This line won't be reached due to the raised error
             except StageOutError as ex:
                 msg = "Attempt %s to stage out failed.\n" % retryCount
-                msg += "Automatically retrying in %s secs\n " % self.retryPause
                 msg += "Error details:\n%s\n" % str(ex)
                 logging.error(msg)
                 if retryCount == self.numRetries:
-                    #  //
-                    # // last retry, propagate exception
-                    # //
-                    raise ex
+                    # Last retry, propagate the information outside of the for loop
+                    stageOutEx = ex
+                msg += "Automatically retrying in %s secs\n " % self.retryPause 
```

Comment:
I would suggest keeping the word stage out in this message. So something like "Automatically retrying stage out in ..."